pkg/validations/extendedversion.go

```diff
@@ -0,0 +1,36 @@
+package validations
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/clients/kubernetes"
+	"github.com/aws/eks-anywhere/pkg/constants"
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/signature"
+	"github.com/aws/eks-anywhere/release/api/v1alpha1"
+)
+
+// ValidateExtendedK8sVersionSupport validates all the validations needed for the support of extended kubernetes support.
+func ValidateExtendedK8sVersionSupport(_ context.Context, _ *anywherev1.Cluster, bundle *v1alpha1.Bundles, _ kubernetes.Client) error {
+	// Validate EKS-A bundle has not been modified by verifying the signature in the bundle annotation
+	if err := ValidateBundleSignature(bundle); err != nil {
+		return fmt.Errorf("validating bundle signature: %v", err)
+	}
+	return nil
+}
+
+// ValidateBundleSignature validates bundles signature with the KMS public key.
+func ValidateBundleSignature(bundle *v1alpha1.Bundles) error {
+	valid, digest, _, err := signature.ValidateSignature(bundle, constants.KMSPublicKey)
+	if err != nil {
+		return err
+	}
+	if !valid {
+		return fmt.Errorf("signature on the bundle is invalid, error: %v, digest: %s", err, base64.StdEncoding.EncodeToString(digest[:]))
+	}
+	logger.Info("Signature on the bundle is valid.")
```

Comment:
i recommend not logging in a function in a package like this. I recommend letting consumers/callers of the function do the logging. Also, this this looks like a global logger. Where is it even defined?

Reply:
Agree, I added it as a part of my debugging and forgot to remove. I have removed this log.